File: app/src/main/java/br/com/jadlog/signature/MainActivity.java
package br.com.jadlog.signature;

import android.content.Intent;
import android.support.v7.app.ActionBar;
import android.support.v7.widget.Toolbar;
import android.util.Log;
import android.view.KeyEvent;

import org.androidannotations.annotations.AfterViews;
import org.androidannotations.annotations.Click;
import org.androidannotations.annotations.EActivity;
import org.androidannotations.annotations.OptionsItem;
import org.androidannotations.annotations.OptionsMenu;
import org.androidannotations.annotations.ViewById;

import br.com.jadlog.signature.ui.SignatureView;

@EActivity(R.layout.activity_main)
@OptionsMenu(R.menu.menu_main)
public class MainActivity extends CordovaActivity {
    @ViewById(R.id.toolbar)
    protected Toolbar toolbar;

    protected ActionBar actionBar;

    @ViewById(R.id.signature)
    protected SignatureView signatureView;

    @AfterViews
    protected void afterViews() {
        // toolbar
        setSupportActionBar(toolbar);
        actionBar = getSupportActionBar();

        if (actionBar != null) {
            actionBar.setTitle("");
        }
    }

    @Click(R.id.btnClear)
    protected void clear() {
        signatureView.clear();
    }

    @Click(R.id.btnSave)
    protected void save() {
        String hash = signatureView.getHash();

        Log.d("Signature","HASH: " + hash);

        Intent intent = new Intent(this, FinallyActivity_.class);
               intent.putExtra("HASH", hash);

        startActivity(intent);
        finish();
        overridePendingTransition( R.anim.righttoleft, R.anim.stable );
    }

    @OptionsItem(R.id.action_close)
    protected void close() {
        finish();
    }

    @Override
    public boolean onKeyDown(int keyCode, KeyEvent event) {
        if (keyCode == KeyEvent.KEYCODE_BACK) { finish(); return false; }
        else                                  { return super.onKeyDown(keyCode, event); }
    }
}

File: app/src/main/java/br/com/jadlog/signature/view/MainActivity.kt
package br.com.jadlog.signature.view

import android.content.Intent
import android.graphics.Bitmap
import android.view.KeyEvent
import android.view.Menu
import android.view.MenuItem
import androidx.appcompat.app.AlertDialog
import kotlinx.android.synthetic.main.main_activity.*

import org.koin.android.viewmodel.ext.android.viewModel

import br.com.jadlog.signature.R
import br.com.jadlog.signature.ui.OnAssinaturaListener

class MainActivity : BaseActivity(R.layout.main_activity) {
    private val viewModel: MainViewModel by viewModel()

    override fun initViews() {
        assinaturaComponent.setOnAssinaturaListener(
                object : OnAssinaturaListener {
                    override fun onBitmap(bitmap: Bitmap?) {
                    }

                    override fun onByteArray(byteArray: ByteArray?) {
                        assinaturaComponent.hide()
                        confirm(byteArray)
                    }

                    override fun onHash(hash: String?) {
                    }
                }
        )

        btnAssinatura.setOnClickListener { open() }
    }

    override fun initViewModel() {}

    private fun confirm(hash: ByteArray?) {
        if (hash == null) {
            msg()
            return
        }
        val intent = Intent(this, FinallyActivity::class.java)
            intent.putExtra("HASH", hash)
        startActivity(intent)
        finish()
        overridePendingTransition(R.anim.righttoleft, R.anim.stable)
    }

    private fun open() {
        assinaturaComponent.show()
    }

    //**********************************************************************************************
    // Menu Principal
    //**********************************************************************************************
    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.action_close -> finish()
            R.id.action_open  -> open()
        }
        return true
    }

    //**********************************************************************************************
    private fun msg() {
        // abre a janela
        val dialog = AlertDialog.Builder(this, R.style.AlertDialogThemeMsg)
            dialog.setTitle(R.string.signature_error_title)
            dialog.setMessage(R.string.signature_error_msg)
            dialog.setCancelable(false)
            dialog.setNeutralButton("OK") { dialog, whichButton -> }

        // show the Dialog
        dialog.create().show()
    }

    override fun onKeyDown(keyCode: Int, event: KeyEvent): Boolean {
        return if (keyCode == KeyEvent.KEYCODE_BACK) {
            finish()
            false
        }
        else {
            super.onKeyDown(keyCode, event)
        }
    }
}